nvtabular/io/parquet.py

```diff
@@ -109,6 +132,458 @@ def to_ddf(self, columns=None):
+        # Also check if any row groups are larger than recommended.
+        if max_rg_size > row_group_max_size:
+            # One or more row-groups are above the "required" limit
+            msg = (
+                f"Excessive row_group size ({max_rg_size}) detected in file "
+                f"{max_rg_size_path}. Please use the regenerate_dataset utility "
+                f"to rewrite your dataset."
+            )
+            warnings.warn(msg)
+        else:
+            # The only way size_valid==True is if we get here
+            size_valid = True
+
+        # Check if any files are smaller than the desired size.
+        # We only warn if there are >1 files in the dataset.
+        for path, size in file_sizes.items():
+            if size < file_min_size and len(pa_dataset.pieces) > 1:
+                msg = (
+                    f"File {path} is smaller than the desired dataset "
+                    f"partition size ({self.part_size}). Consider using the "
+                    f"regenerate_dataset utility to rewrite your dataset with a smaller "
+                    f"number of (larger) files."
+                )
+                warnings.warn(msg)
+                size_valid = False
+
+        # If the _metadata file is missing, we need to write
+        # it (or inform the user that it is missing)
+        if not metadata_file_exists:
+            if add_metadata_file:
+                # Write missing _metadata file
+                fs = self.fs
+                metadata_path = fs.sep.join([root_dir, "_metadata"])
+                with fs.open(metadata_path, "wb") as fil:
+                    metadata.write_metadata_file(fil)
+                meta_valid = True
+            else:
+                # Inform user that the _metadata file is missing
+                msg = (
+                    "For best performance with NVTabular, there should be a "
+                    "global _metadata file located in the root directory of the "
+                    "dataset. Please pass add_metadata_file=True to add the "
+                    "missing file."
+                )
+                warnings.warn(msg)
+                if require_metadata_file:
+                    meta_valid = False
+
+        # Return True if we have a parquet dataset with a _metadata file (meta_valid)
+        # and the row-groups and file are appropriate sizes (size_valid)
+        return meta_valid and size_valid
+
+    @classmethod
+    def regenerate_dataset(
+        cls,
+        dataset,
+        output_path,
+        columns=None,
+        file_size=None,
+        part_size=None,
+        cats=None,
+        conts=None,
+        labels=None,
+        storage_options=None,
+    ):
+        """Regenerate an NVTabular Dataset for efficient processing.
+
+        Example Usage::
+
+            dataset = Dataset("/path/to/data_pq", engine="parquet")
+            dataset.regenerate_dataset(
+                out_path, part_size="1MiB", file_size="10MiB"
```

Comment:
Our recommendations here are around 128MB for part_size and 10GB or so for file_size - should we reflect that here?